src/modules/agora/AgoraClient.ts

```diff
@@ -108,6 +108,33 @@ export class AgoraClient implements FullNodeAPI
                 });
         });
     }
+
+    /**
+     * Request the latest PreImageInfo for an enrollment key
+     *
+     * @param enroll_key Hash of the UTXO used by the validator
+     * @returns A `PreImageInfo` matching this `enroll_key`
+     */
+    public getPreimage (enroll_key: Hash): Promise<PreImageInfo>
```

Comment:
Do you want to change the method of collecting pre-image data from `put` to `get` request?

Reply:
Or is it just for testing?

Reply:
No, we keep Agora's push approach, this is here for a fallback.